pandas/tests/extension/base/methods.py

```diff
@@ -125,7 +125,11 @@ def test_sort_values(self, data_for_sorting, ascending, sort_by_key):
         result = ser.sort_values(ascending=ascending, key=sort_by_key)
         expected = ser.iloc[[2, 0, 1]]
         if not ascending:
-            expected = expected[::-1]
+            # GH 35922. Expect stable sort
+            if ser.nunique() == 2:
+                expected = ser.iloc[[0, 1, 2]]
+            else:
+                expected = ser.iloc[[1, 0, 2]]
```

Comment:
Some of our parametrizations in `data_for_sorting` contain dupliates and some don't. Since they are all structured the same way, we can detect different expected sort orders by checking the number of unique values. If it looks ugly to you, we could split `data_for_sorting` for this test.

Reply:
ok this is fine